pkg/oomstore/revision.go

```diff
@@ -24,6 +25,85 @@ func (s *OomStore) GetRevisionBy(ctx context.Context, groupID int, revision int6
 	return s.metadata.GetRevisionBy(ctx, groupID, revision)
 }
 
+func (s *OomStore) CreateRevision(ctx context.Context, opt metadata.CreateRevisionOpt) (int, string, error) {
+	var revisionID int
+	var snapshotTable string
+	var dummyRevision *types.Revision
+
+	if err := s.metadata.WithTransaction(ctx, func(c context.Context, db metadata.DBStore) error {
+		_, err := s.metadata.GetRevisionBy(ctx, opt.GroupID, 0)
+		if err != nil {
+			if !errdefs.IsNotFound(err) {
+				return err
+			}
+
+			if _, _, err = s.metadata.CreateRevision(ctx, metadata.CreateRevisionOpt{
+				Revision:    0,
+				GroupID:     opt.GroupID,
+				Description: "dummy revision will be used at Join and Export",
+			}); err != nil {
+				return err
+			}
+
+			if dummyRevision, err = s.metadata.GetRevisionBy(ctx, opt.GroupID, 0); err != nil {
+				return err
+			}
+		}
+
+		revisionID, snapshotTable, err = s.metadata.CreateRevision(ctx, opt)
+		return err
+	}); err != nil {
+		return 0, "", err
+	}
+
+	if dummyRevision != nil {
+		if err := s.createSnapshotAndCdcTable(ctx, dummyRevision); err != nil {
+			return 0, "", err
+		}
+	}
+
+	return revisionID, snapshotTable, nil
+}
+
+func (s *OomStore) createSnapshotAndCdcTable(ctx context.Context, revision *types.Revision) error {
+	snapshotTable := dbutil.OfflineStreamSnapshotTableName(revision.GroupID, revision.Revision)
+
+	// Create snapshot table in offline store
+	features, err := s.metadata.ListFeature(ctx, metadata.ListFeatureOpt{
+		GroupID: &revision.GroupID,
+	})
+
+	if err = s.offline.CreateTable(ctx, offline.CreateTableOpt{
+		TableName: snapshotTable,
+		Entity:    revision.Group.Entity,
+		Features:  features,
+		TableType: types.TableStreamSnapshot,
+	}); err != nil {
+		return err
+	}
+
+	// Update snapshot_table in feature_group_revision table
+	if err := s.metadata.UpdateRevision(ctx, metadata.UpdateRevisionOpt{
+		RevisionID:       revision.ID,
+		NewSnapshotTable: &snapshotTable,
```

Comment:
should we also set `CdcTable` for the revision?

Reply:
`metadata.UpdateRevision` currently does not support updating the `CdcTable`, I want to change it in the next PR.

Reply:
sounds good!